documentation/bolt_defaults_reference.md.erb

```diff
@@ -0,0 +1,83 @@
+# `bolt-defaults.yaml` options
+
+This page lists the configuration options that are available in the `bolt-defaults.yaml` configuration file,
+which is supported in both the [system-wide and user-level configuration directories](configuring_bolt.md).
+
+<% @opts[:options].each do |option, desc| -%>
+## `<%= option %>`
+
+<%= desc %>
+
+<% if @opts[:defaults][option] -%>
+- **Default:** <%= @opts[:defaults][option] %>
+<% end -%>
+
+<% if @opts[:suboptions][option] -%>
+### Options
+
+This option accepts a map of additional options.
+
+<% if option == 'inventory-config' %>
+<% @opts[:suboptions][option].each do |option, _| -%>
+- `<%= option %>`
+<% end -%>
+
+A detailed description of each option, their default values, and any available sub-options
+can be viewed in [Transport configuration reference](bolt_transports_reference.md).
+<% else %>
+<% @opts[:suboptions][option].each do |option, desc| -%>
+#### `<%= option %>`
+
+<%= desc %>
+<% if @transports.include?(option) -%>
+This option accepts a map of additional options. You can view the available options in [Transport configuration reference](bolt_transports_reference.md#<%= option %>).
+<% end %>
+
+<% if @opts[:defaults][option] -%>
+- **Default:** <%= @opts[:defaults][option] %>
+<% end %>
+<% end -%>
+<% end %>
+<% end -%>
+<% end -%>
+
+## Example file
+
+```yaml
+color: true
+compile-concurrency: 8
+concurrency: 100
+format: human
```

Comment:
These are also just the defaults...is this just being generated somehow? If not I think it'd be nice to have them be non-defaults to make it a little more real, but if they are generated that's probably fine.

Reply:
This is manually created, but as part of the work for #1888 I'm looking at having the files auto-generated. That work has non-default values used. I'll update this to use non-defaults as well.

Reply:
Discussed in slack - this will get fixed up as part of #1888